test/metabase/util/honey_sql_2_extensions_test.clj

```diff
@@ -8,6 +8,14 @@
    [metabase.util.honey-sql-2-extensions :as h2x]
    [metabase.util.honeysql-extensions :as hx]))
 
+;; HACK: Workaround for Kondo incorrectly type-checking the original function. (It's a mu/defn-ed function, which
+;; we're linting as if it were defined with schema.core/defn. So Kondo is treating the Malli types as Schema types and
+;; getting confused.
+(defn- with-database-type-info
+  [hsql type-string]
+  #_{:clj-kondo/ignore [:type-mismatch]}
```

Comment:
I think the best quick fix for this is using `clj-kondo.lint-as/def-catch-all`, like we do with defendpoint. That way we won't need to go back and un-ignore anything once the mu/defn hook is in place.

[I made sure that works here](https://github.com/metabase/metabase/commit/698cc6c8af8f7c726f8aff5f0b1a2186a060dd13).

What do you think?